server/src/main/java/org/elasticsearch/cluster/ClusterInfoSimulator.java

```diff
@@ -39,10 +39,10 @@ public void simulate(ShardRouting shard) {
             if (shard.relocatingNodeId() != null) {
                 // relocation
                 modifyDiskUsage(shard.relocatingNodeId(), getShardPath(shard.relocatingNodeId(), mostAvailableSpaceUsage), size);
-                modifyDiskUsage(shard.currentNodeId(), getShardPath(shard.currentNodeId(), leastAvailableSpaceUsage), -size);
+                modifyDiskUsage(shard.currentNodeId(), getShardPath(shard.currentNodeId(), mostAvailableSpaceUsage), -size);
```

Comment:
I think this means that we never modify the least available space usage and that we can thus simplify `modifyDiskUsage` to only work on `mostAvailableSpaceUsage`?

Reply:
Can we then modify `getShardPath` to work on `mostAvailableSpaceUsage` always rather than through a parameter?